configure.ac

```diff
@@ -878,7 +865,7 @@ AS_IF([test "x$squid_opt_use_esi" != "xno" -a "x$with_xml2" != "xno"],[
 AM_CONDITIONAL(ENABLE_LIBXML2,[test "x$LIBXML2_LIBS" != "x"])
 AC_SUBST(LIBXML2_LIBS)
 
-AS_IF([test "x$squid_opt_use_esi" = "xyes"],[
+AS_IF([test "x$LIBEXPAT_LIBS" != "x" -o "x$squid_opt_use_esi" = "xyes"],[
   AS_IF([test "x$LIBXML2_LIBS" = "x" -a "x$LIBEXPAT_LIBS" = "x"],[
     AC_MSG_ERROR([ESI processor requires libxml2 or libexpat])
   ])
```

Comment:
On the next line (I cannot attach a change request there), Squid enables ESI support. Thus, this PR introduces a bug: If libexpat is found, Squid enables ESI support despite `--disable-esi`.

FWIW, I recommend merging #1727 changes into this PR so that we can write this part of the code once (and correctly) instead of trying to adjust it twice, once in each minimal (but necessarily conflicting from git point of view) PR. I do not insist on that strategy.

Reply:
resolved by changes now in master branch